pkg/cloud/ovirt/machine/actuator.go

```diff
@@ -376,10 +376,65 @@ func (actuator *OvirtActuator) reconcileConditions(
 	return conditions
 }
 
+
+// validateMachine execute several validations, example: check if user provided in
+// the machineset UserDataSecret, CredentialsSecret etc. If error exists, it will
+// return apierrors.MachineError
 func (actuator *OvirtActuator) validateMachine(machine *machinev1.Machine, config *ovirtconfigv1.OvirtMachineProviderSpec) *apierrors.MachineError {
-	return nil
+
+	errmsg := "error creating oVirt instance:"
+
+	// UserDataSecret
+	if config.UserDataSecret == nil {
+		return apierrors.CreateMachine(
+			fmt.Sprintf("%s UserDataSecret must be provided!", errmsg))
+	} else {
+		if config.UserDataSecret.Name == "" {
+			return apierrors.CreateMachine(
+				fmt.Sprintf("%s UserDataSecret *Name* must be provided!", errmsg))
+		}
+	}
+
+	// CredentialsSecret
+	if config.CredentialsSecret == nil {
+		return apierrors.CreateMachine(
+			fmt.Sprintf("%s CredentialsSecret must be provided!", errmsg))
+	} else {
+		if config.CredentialsSecret.Name == "" {
+			return apierrors.CreateMachine(
+				fmt.Sprintf("%s CredentialsSecret *Name* must be provided!", errmsg))
+		}
+	}
+
+	// If InstanceTypeId is NOT SET users can set MEM and CPU,
+	// all memory and cpu variables will be ignored.
+	if len(config.InstanceTypeId) == 0 {
```

Comment:
This you should have logic here to check that only one of them is available and fail on invalid configuration in case non or both of them are provided

Reply:
I want to test that only one of them is used, meaning if a user added InstanceTypeId and CPU/MEM then he should get an invalid configuration